lib/admin/reducers/servers.js

```diff
@@ -19,6 +19,18 @@ const servers = (state: AdminServersState = defaultState, action: Action): Admin
         isFetching: { $set: false },
         data: { $set: action.payload }
       })
+    case 'RECEIVE_SERVER':
+      const serverData = action.payload
+      if (state.data) {
+        const serverIdx = state.data.findIndex(
+          server => server.id === serverData.id
+        )
+        return update(state, {
+          isFetching: { $set: false },
+          data: { [serverIdx]: { $set: action.payload } }
```

Comment:
Not necessarily needed, but seems like an opportunity to reuse the serverData variable?